frontend/public/components/dashboard/with-dashboard-resources.tsx

```diff
@@ -140,7 +140,7 @@ export const withDashboardResources = <P extends DashboardItemProps>(
 
       watchK8sResource: WatchK8sResource = (resource) => {
         this.setState((state: WithDashboardResourcesState) => ({
-          k8sResources: [...state.k8sResources, resource],
+          k8sResources: [...state.k8sResources, { ...resource, optional: true }],
```

Comment:
What does it look like in the UI if these resources fail?

Reply:
We would be stuck in loading state because the resource is simply omitted from injected `resources` object. Its better than failing miserably but still far from ideal behavior.

Reply:
Is there a reason we don't have flags for these resources and only add them to the resources list if they exist in the cluster?